From amb@gedanken.demon.co.uk Tue Aug 21 21:05:50 2007
From: Andrew M. Bishop <amb@gedanken.demon.co.uk>
Date: 21 Aug 2007 19:08:56 +0100
Subject: USB: ftdi_sio: Handle FT232RL devices like FT232BM devices
To: Greg KH <greg@kroah.com>
Cc: Bill Ryder <bryder@sgi.com>,  Kuba Ober <kuba@mareimbrium.org>
Message-ID: <vohcmswyiv.fsf_-_@gedanken.demon.co.uk>


Handle the FT232RL device type in exactly the same way as FT232BM
devices (FT232RL detection was added around kernel 2.6.20 but not code
for handling it).

Signed-off-by: Andrew M. Bishop <amb@gedanken.demon.co.uk>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>


---
 drivers/usb/serial/ftdi_sio.c |    5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

--- a/drivers/usb/serial/ftdi_sio.c
+++ b/drivers/usb/serial/ftdi_sio.c
@@ -1168,7 +1168,9 @@ static void remove_sysfs_attrs(struct us
 	/* XXX see create_sysfs_attrs */
 	if (priv->chip_type != SIO) {
 		device_remove_file(&port->dev, &dev_attr_event_char);
-		if (priv->chip_type == FT232BM || priv->chip_type == FT2232C) {
+		if (priv->chip_type == FT232BM ||
+		    priv->chip_type == FT2232C ||
+		    priv->chip_type == FT232RL) {
 			device_remove_file(&port->dev, &dev_attr_latency_timer);
 		}
 	}
@@ -2101,6 +2103,7 @@ static int ftdi_tiocmget (struct usb_ser
 	case FT8U232AM:
 	case FT232BM:
 	case FT2232C:
+	case FT232RL:
 		/* the 8U232AM returns a two byte value (the sio is a 1 byte value) - in the same
 		   format as the data returned from the in point */
 		if ((ret = usb_control_msg(port->serial->dev,
